src/ProjectNeon/Assets/Features/Effects/EffectOnTurnStart.cs

```diff
@@ -0,0 +1,30 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Effects that trigger on Turn Start.
+ */
+public class EffectOnTurnStart : Effect
+{
+    private Effect _effect;
+    private Member _performer;
+    private Target _effectTarget;
+
+    public EffectOnTurnStart(Effect origin)
+    {
+        _effect = origin;
+    }
+
+    public void Apply(Member source, Target target)
+    {
+        _performer = source;
+        _effectTarget = target;
+        BattleEvent.Subscribe<TurnStart>((turnStart) => Execute(), this);
```

Comment:
This must get cleaned up to prevent Memory Leak.

Need unsubscribe logic.

Reply:
@SilasReinagel Please see the other one about unsubscriving events